Simulator/tft_simulator.py

```diff
@@ -169,20 +169,22 @@ def step(self, action):
             return
         action = np.asarray(action)
         if action.ndim == 0:
-            self.step_function.action_controller(action, self.PLAYERS[self.agent_selection], self.PLAYERS,
+            # self.step_function.action_controller(action, self.PLAYERS[self.agent_selection], self.PLAYERS,
+            #                                      self.agent_selection, self.game_observations)
+            self.step_function.single_step_action_controller(action, self.PLAYERS[self.agent_selection], self.PLAYERS,
```

Comment:
One of the restrictions of petting zoo is that the actions need to be integers. You can't send a bunch of probabilities to the step function with the current set up. If you change the action_space type to something that is no discrete. To a continuous type of action space, then you can probably get away with this type of action. The problem there is it completely breaks the MCTS because MCTS requires non-continuous action spaces in it's default setting. If you want to use continuous action spaces, we will need to implement the paper for complex action spaces. If we are doing that, we might as well use a multi-dimensional action space.

Reply:
The idea was for the step function to be able to be used with any model-free agent, but the action is handled in a discrete way inside, so np.argmax of discrete = discrete